File: week1/rlagkswn00/BOJ_1018.py
#
# Title: BOJ 1018 체스판 다시 칠하기
# Theory :
# Date: 23.07.31
#
import sys
input = sys.stdin.readline

board = []
wb = [
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW"
]
bw = [
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB"
]


def bwCnt(l, m):
    cnt = 0
    for i in range(8):
        for j in range(8):
            if (board[l + i][m + j] != bw[i][j]):
                cnt += 1
    return cnt


def wbCnt(l, m):
    cnt = 0
    for i in range(8):
        for j in range(8):
            if (board[l + i][m + j] != wb[i][j]):
                cnt += 1
    return cnt


min_val = 9999999
x, y = map(int, input().split())

for i in range(x):
    board.append(input())

for i in range(0, x-8 + 1):
    for j in range(0, y-8 + 1):
        tmp = min(wbCnt(i, j), bwCnt(i, j))
        if (tmp < min_val):
            min_val = tmp
print(min_val)

File: week1/chaeyoung/BOJ_1018.cpp
#include <iostream>
#include <algorithm>
#define MAX 51
using namespace std;

int N, M, min_val = 100;
char arr[MAX][MAX];

string WB[8] = {
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW"
};

string BW[8] = {
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB",
        "BWBWBWBW",
        "WBWBWBWB"
};

int start_W(int x, int y){
    int cnt = 0;
    for(int i = 0; i < 8; i++){
        for(int j = 0; j < 8; j++){
            if(arr[x+i][y+j] == WB[i][j]){
                cnt++;
            }
        }
    }
    return cnt;
}

int start_B(int x, int y){
    int cnt = 0;
    for(int i = 0; i < 8; i++){
        for(int j = 0; j < 8; j++){
            if(arr[x+i][y+j] == BW[i][j]){
                cnt++;
            }
        }
    }
    return cnt;
}

int main()
{
    cin >> N >> M;
    for(int i = 0; i < N; i++){
        for(int j = 0; j < M; j++){
            cin >> arr[i][j];
        }
    }
    for(int i = 0; i+8 <= N; i++){
        for(int j = 0; j+8 <= M; j++){
            int tmp;
            tmp = min(start_B(i,j),start_W(i,j));
            if(tmp < min_val) {
                min_val = tmp;
            }
        }
    }
    cout << min_val;
    return 0;
}